src/test/java/teammates/test/cases/datatransfer/FeedbackQuestionAttributesTest.java

```diff
@@ -95,6 +96,29 @@ public void testValueOf() throws InvalidParametersException, EntityAlreadyExists
         assertEquals(qn.getId(), feedbackQuestionAttributes.getFeedbackQuestionId());
     }
 
+    @Test
+    public void testValueOf_withDifferentTextQuestions() throws InvalidParametersException {
+        ______TS("single word text, should deserialize as plain text");
+        FeedbackQuestionsDb db = new FeedbackQuestionsDb();
+        FeedbackQuestion qn = db.createEntityWithoutExistenceCheck(getNewFeedbackQuestionAttributes());
+        qn.setQuestionText("singleWord");
+
+        FeedbackQuestionAttributes fqa = FeedbackQuestionAttributes.valueOf(qn);
+        assertEquals("singleWord", fqa.questionDetails.getQuestionText());
+        assertEquals(0, ((FeedbackTextQuestionDetails) fqa.questionDetails).getRecommendedLength());
```

Comment:
Add another test `legacy data: plain text: multiple words, should deserialise correctly`